DrawerLayout.js

```diff
@@ -198,9 +196,22 @@ export default class DrawerLayout extends Component<PropType, StateType> {
       }
     );
 
+    const gestureOptions = {
+      useNativeDriver: props.useNativeAnimations
+    };
+
+    if( this.props.onDrawerSlide ) {
+      gestureOptions.listener = ev => {
+        const translationX = Math.floor(Math.abs(ev.nativeEvent.translationX));
+        const position = translationX / this.state.containerWidth;
+
+        this.props.onDrawerSlide(position);
+      };
+    }
+
     this._onGestureEvent = Animated.event(
       [{ nativeEvent: { translationX: dragXValue, x: touchXValue } }],
-      { useNativeDriver: props.useNativeAnimations }
+      gestureOptions
```

Comment:
I'm not sure extracting this object is more readable than older code.

Reply:
How would you prefer to conditionally add the listener function?

Reply:
Oh, I missed this assignment. 
This could be written with spread and `&&` operator, but i don't think it's more readable than the current solution.